packages/desktop-client/src/components/modals/CreateLocalAccountModal.tsx

```diff
@@ -23,6 +25,21 @@ type CreateLocalAccountProps = {
   actions: BoundActions;
 };
 
+/*
+ * Ensure the name is not empty and does not already exist.
+ */
+const validateName = (name: string, accounts: AccountEntity[]) => {
+  if (!name) {
+    return { error: true, message: 'Name is required' };
+  }
+
+  if (accounts.some(account => account.name === name)) {
+    return { error: true, message: `Name: “${name}” already exists.` };
+  }
+
+  return { error: false, message: '' };
```

Comment:
<!-- Nitpicks are small, trivial, but necessary changes. Distinguishing nitpick comments significantly helps direct the reader's attention to comments requiring more involvement.  -->
**🥜 nitpick:** ‏what if we always returned a `string` in this util? Then we could re-use it in both this modal and the account page. Slightly less duplication of logic.

Reply:
I agree.  If this just returns a string (empty string for no error) then you can reuse it above, and you can simplify the code below as you can just check `if (nameError)` instead of `if (nameError.error)`.  You can also apply the other suggestion Mattis has.